Returning `Interval` instead of tuples in distributions' Support members

valbers opened the pull request:
* Fixes #292 

This pull request introduces **breaking changes**.

Although these are just some small changes, they do represent breaking changes as the public contract of the `Support` members are changed.
The `Support`s of the exponential distribution and of the discrete negative binomial have also been fixed (were previously closed intervals, but they should have the right side "open").

**Check list**

 - [x] The project builds without problems on my machine

 - [x] Adapted unit tests regarding the added features

valbers committed:
Using `Interval` instead of tuples in distr.'s supports

valbers committed:
Fixing support range of exp. distr. and added better fitting...

  Better fitting Support member without the superfluous lambda. Marking
the other Support member as obsolete.

valbers committed:
Using `Interval` instead of tuple in the `Support` of Negative...

  ... of NegativeBinomial

bvenn commented on src/FSharp.Stats/Distributions/Continuous/F.fs:
It seems like the support of the F distribution is dependent on the degrees of freedom used (dof1). At least [here](https://en.wikipedia.org/wiki/F-distribution) it says that its either [0,inf) or (0,inf) depending on dof1 beeing 1 or not. Do you have some thoughts if its correct to match dof1 beeing 0, or 1 respectively and then return different intervals?

valbers replied:
Good point. Additionally, it seems that `dof2` is irrelevant for determining the value of `Support`, so this parametert can be removed too.

bvenn commented on src/FSharp.Stats/Distributions/Continuous/Gamma.fs:
I assume the description is an artifact from the exponential distribution (basically it says so in the /// comment) and the interval should be (0,inf). The same may apply to LogNormal

Reference
- https://en.wikipedia.org/wiki/Gamma_distribution

bvenn commented on src/FSharp.Stats/Distributions/Continuous/Normal.fs:
Because interval is -infinity, the interval should be open. Sorry about the misleading comments above each distribution support. Could you exchange exponential with the appropriate distribution discription for each distribution? If not, no worries I can take care of if, just let me know.

valbers replied:
Sure. Sorry, it seems like I missed this one. But you're right, some of the current comments are not helping 😄

bvenn reviewed (changes requested):
Thanks for the contribution :rocket: 

Sadly the comments above the support members seems to be artifacts from copying it from the exponential distribution. Could you please update the /// description and check the comments I made in this PR?

valbers committed:
Corrected `Support` of the F distr. Also: ...

  Removed `dof2` as a parameter of `Support`, since it plays no role
there.
  Also added a helper module to the F distribution file with a helper pure function. I hope it's ok.

valbers committed:
Corrected comment on `Support` of Gamma distr.

valbers committed:
Correcting comment of Gamma `Support` again :)

valbers committed:
Correcting `Support` of the LogNormal distr.

valbers committed:
corrected failure message of `assertValidDof`

valbers committed:
Correcting `Support` of the Gamma distr.
